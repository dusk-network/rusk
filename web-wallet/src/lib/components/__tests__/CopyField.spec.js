import { afterEach, describe, expect, it, vi } from "vitest";
import { cleanup, fireEvent, render } from "@testing-library/svelte";
import { CopyField } from "..";

Object.assign(navigator, {
  clipboard: {
    writeText: vi.fn().mockResolvedValue(""),
  },
});

describe("CopyField", () => {
  const baseProps = {
    disabled: false,
    displayValue: "1,234,567",
    name: "Sample Information",
    rawValue: "1234567",
  };

  const baseOptions = {
    props: baseProps,
    target: document.body,
  };

  afterEach(cleanup);

  it("renders the CopyField component", () => {
    const { container } = render(CopyField, baseOptions);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should pass additional class names and attributes to the rendered element", () => {
    const props = {
      ...baseProps,
      className: "foo bar",
      id: "some-id",
    };
    const { container } = render(CopyField, { ...baseOptions, props });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the CopyField component with the copy button disabled", () => {
    const { container, getByRole } = render(CopyField, {
      ...baseOptions,
      props: { ...baseProps, disabled: true },
    });

    const copyButton = getByRole("button");

    expect(copyButton).toBeDisabled();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("copies the raw value on pressing the copy button", async () => {
    const { getByRole } = render(CopyField, baseOptions);

    const copyButton = getByRole("button");

    await fireEvent.click(copyButton);

    expect(navigator.clipboard.writeText).toHaveBeenCalledWith("1234567");
  });
});
