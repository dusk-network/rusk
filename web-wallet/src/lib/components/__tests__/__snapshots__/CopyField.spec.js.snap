// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CopyField > renders the CopyField component 1`] = `
<div
  class="copy-field"
>
  <input
    class="dusk-textbox dusk-textbox-text copy-field__content"
    readonly=""
    type="text"
  />
  
   
  <button
    aria-label="Copy Address"
    class="dusk-button dusk-button--type--button dusk-button--variant--primary dusk-button--size--default dusk-icon-button copy-field__button"
    data-tooltip-id="main-tooltip"
    data-tooltip-text="Copy to clipboard"
    type="button"
  >
    <svg
      class="dusk-icon dusk-icon--size--default dusk-button__icon"
      role="graphics-symbol"
      viewBox="0 0 24 24"
    >
      <path
        d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"
      />
    </svg>
    
     
    
  </button>
</div>
`;

exports[`CopyField > renders the CopyField component with the copy button disabled 1`] = `
<div
  class="copy-field"
>
  <input
    class="dusk-textbox dusk-textbox-text copy-field__content"
    readonly=""
    type="text"
  />
  
   
  <button
    aria-label="Copy Address"
    class="dusk-button dusk-button--type--button dusk-button--variant--primary dusk-button--size--default dusk-icon-button copy-field__button"
    data-tooltip-id="main-tooltip"
    data-tooltip-text="Copy to clipboard"
    disabled=""
    type="button"
  >
    <svg
      class="dusk-icon dusk-icon--size--default dusk-button__icon"
      role="graphics-symbol"
      viewBox="0 0 24 24"
    >
      <path
        d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"
      />
    </svg>
    
     
    
  </button>
</div>
`;

exports[`CopyField > should pass additional class names and attributes to the rendered element 1`] = `
<div
  class="copy-field foo bar"
  id="some-id"
>
  <input
    class="dusk-textbox dusk-textbox-text copy-field__content"
    readonly=""
    type="text"
  />
  
   
  <button
    aria-label="Copy Address"
    class="dusk-button dusk-button--type--button dusk-button--variant--primary dusk-button--size--default dusk-icon-button copy-field__button"
    data-tooltip-id="main-tooltip"
    data-tooltip-text="Copy to clipboard"
    type="button"
  >
    <svg
      class="dusk-icon dusk-icon--size--default dusk-button__icon"
      role="graphics-symbol"
      viewBox="0 0 24 24"
    >
      <path
        d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"
      />
    </svg>
    
     
    
  </button>
</div>
`;
